### PR TITLE
EZP-27359: Fully integrate FOSJsRoutingBundle

### PR DESCRIPTION
**Pull request by lolautruche**
> https://jira.ez.no/browse/EZP-27359

* [x] Make FOSJsRouting compatible with URI SiteAccesses (only if FOSJsRoutingBundle is present)
* [x] Expose _ez_content_view route correctly

### CODE REVIEW

**Comment by lolautruche** [comment]
Review ping @bdunogier @andrerom @dpobel

**Comment by lolautruche** [comment]
@andrerom @bdunogier Could you please restart Travis ? It seems it had a few issues

**Comment by andrerom** [comment]
Restarted, but don't think that will help:
```
  Problem 1
    - Installation request for ezsystems/ezpublish-kernel dev-tmp_ci_branch -> satisfiable by ezsystems/ezpublish-kernel[dev-tmp_ci_branch].
    - ezsystems/hybrid-platform-ui dev-master requires ezsystems/ezpublish-kernel  ^6.9@dev -> satisfiable by ezsystems/ezpublish-kernel[6.11.x-dev, 6.10.x-dev, 6.9.x-dev].
    - don't install ezsystems/ezpublish-kernel 6.10.x-dev|don't install ezsystems/ezpublish-kernel dev-tmp_ci_branch
    - don't install ezsystems/ezpublish-kernel 6.9.x-dev|don't install ezsystems/ezpublish-kernel dev-tmp_ci_branch
    - don't install ezsystems/ezpublish-kernel 6.11.x-dev|don't install ezsystems/ezpublish-kernel dev-tmp_ci_branch
    - don't install ezsystems/ezpublish-kernel dev-tmp_ci_branch|don't install ezsystems/ezpublish-kernel 6.11.x-dev
    - Installation request for ezsystems/hybrid-platform-ui dev-master as 1.0.x-dev -> satisfiable by ezsystems/hybrid-platform-ui[dev-master].
```

File: eZ/Bundle/EzPublishCoreBundle/Routing/JsRouting/ExposedRoutesExtractor.php [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+/**
+ * @license For full copyright and license information view LICENSE file distributed with this source code.
+ */
+namespace eZ\Bundle\EzPublishCoreBundle\Routing\JsRouting;
+
+use eZ\Publish\Core\MVC\Symfony\SiteAccess;
+use FOS\JsRoutingBundle\Extractor\ExposedRoutesExtractorInterface;
+use Symfony\Component\HttpFoundation\Request;
+
+class ExposedRoutesExtractor implements ExposedRoutesExtractorInterface
```
Comment (bdunogier) [inline eZ/Bundle/EzPublishCoreBundle/Routing/JsRouting/ExposedRoutesExtractor.php]: A lil' bit of PhpDoc explaining why we redefine an extractor wouldn't hurt.

**Comment by bdunogier** [comment]
About Travis, I have opened https://github.com/ezsystems/ezpublish-kernel/pull/2026 against 7.0 to test fixes for it. Aliasing the tested kernel as 6.99.x-dev (as in ezplatform 2.0 composer.json) works, it seems.

File: eZ/Bundle/EzPublishCoreBundle/Routing/JsRouting/ExposedRoutesExtractor.php (hunk 1 above)
Comment (lolautruche) [inline eZ/Bundle/EzPublishCoreBundle/Routing/JsRouting/ExposedRoutesExtractor.php]: Updated

File: eZ/Bundle/EzPublishCoreBundle/DependencyInjection/EzPublishCoreExtension.php [hunk 2]
```diff
@@ -101,6 +101,9 @@ public function load(array $configs, ContainerBuilder $container)
         $loader->load('security.yml');
         // Slots
         $loader->load('slot.yml');
+        if (interface_exists('FOS\JsRoutingBundle\Extractor\ExposedRoutesExtractorInterface')) {
```
Comment (bdunogier) [inline eZ/Bundle/EzPublishCoreBundle/DependencyInjection/EzPublishCoreExtension.php]: I guess it could make sense.

Comment (lolautruche) [inline eZ/Bundle/EzPublishCoreBundle/DependencyInjection/EzPublishCoreExtension.php]: I was unsure of the possibility to get the accurate bundles list from the DIC extension... It's possible from a compiler pass, but the situation is different in the extension

Comment (emodric) [inline eZ/Bundle/EzPublishCoreBundle/DependencyInjection/EzPublishCoreExtension.php]: I had no issues with that approach in TagsBundle.

AFAIK the list of bundles is initialized before the extensions are ran.

Comment (bdunogier) [inline eZ/Bundle/EzPublishCoreBundle/DependencyInjection/EzPublishCoreExtension.php]: @lolautruche can you look into it or should I ?

Comment (lolautruche) [inline eZ/Bundle/EzPublishCoreBundle/DependencyInjection/EzPublishCoreExtension.php]: @bdunogier If you can do it, be my guest :-)

Comment (bdunogier) [inline eZ/Bundle/EzPublishCoreBundle/DependencyInjection/EzPublishCoreExtension.php]: I'm replacing it with `if ($container->hasExtension('fos_js_routing'))`. No objections ?

Comment (bdunogier) [inline eZ/Bundle/EzPublishCoreBundle/DependencyInjection/EzPublishCoreExtension.php]: Done in 4afcf2a.

Comment (emodric) [inline eZ/Bundle/EzPublishCoreBundle/DependencyInjection/EzPublishCoreExtension.php]: :+1:

Comment (dpobel) [inline eZ/Bundle/EzPublishCoreBundle/DependencyInjection/EzPublishCoreExtension.php]: @bdunogier @lolautruche @emodric actually, it seems like this *last minute* change broke the patch `$container->hasExtension('fos_js_routing')` is not true (`$container->getExtensions()` returns an empty array at that stage) while with `interface_exists` this works well. So unless, there's an alternative, I'm going to restore the previous condition.

Comment (dpobel) [inline eZ/Bundle/EzPublishCoreBundle/DependencyInjection/EzPublishCoreExtension.php]: See https://github.com/ezsystems/ezpublish-kernel/pull/2038

**Pull request merged**